src/bigquery.rs

```diff
@@ -53,8 +54,10 @@ pub struct Tag {
     mode: Mode,
 }
 
-impl From<ast::Tag> for Tag {
-    fn from(tag: ast::Tag) -> Self {
+impl Translate<ast::Tag> for Tag {
+    type Error = &'static str;
+
+    fn translate(tag: ast::Tag) -> Result<Self, Self::Error> {
```

Comment:
TODO: `translate` is a thin veneer over `from` and doesn't use the `Error` type at all. We should propagate errors up the tree so we can handle dropping empty structs.